pkg/apis/servicecatalog/v1alpha1/types.go

```diff
@@ -364,7 +364,7 @@ type BindingSpec struct {
 
 	// SecretName is the name of the secret to create in the Binding's
 	// namespace that will hold the credentials associated with the Binding.
-	SecretName string `json:"secretName"`
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
While its not optional in etcd (because we end up defaulting it), it is optional for a client to send it in on the createBinding() request.  So, my understanding of `omitempty` is that its a signal to the json serializer that it doesn't need to send "", it can just skip that property. Which makes sense to me - but perhaps `omitempty` here has influence on other parts of the system we need to take into account??